domains/Links/Controllers/LinksStoreController.php

```diff
@@ -29,6 +31,11 @@ public function __invoke(Request $request): Response
             'tags.*.id' => ['required', 'integer', 'exists:tags'],
         ]);
 
+        throw_unless(
+            Gate::allows('create', [Link::class, $request->input('author_email')]),
+            new UnapprovedLinkLimitReachedException()
+        );
```

Comment:
Maybe this can go before the request validation to streamline the response flow based on priority rules (limit of unapproved links has higher priority than validating an incoming link store request)

Reply:
On second thought, this might have been a stupid request on my end... This forced you to send a default empty string for the author's email part of the policy. I'll rectify it to not bother you with the revert.